platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -797,6 +800,49 @@ public void removeOnStyleImageMissingListener(@NonNull OnStyleImageMissingListen
     mapChangeReceiver.removeOnStyleImageMissingListener(listener);
   }
 
+  /**
+   * Set a callback that's invoked when map needs to release unused image resources.
+   *
+   * @param listener The callback that's invoked when map needs to release unused image resources
+   */
+  public void addOnRemoveUnusedStyleImagesListener(@NonNull OnRemoveUnusedStyleImagesListener listener) {
+    mapChangeReceiver.addOnRemoveUnusedStyleImagesListener(listener);
+    removeDefaultRemoveUnusedStyleImagesListener();
+  }
+
+  /**
+   * Removes a callback that's invoked when map needs to release unused image resources.
+   *
+   * @param listener The callback that's invoked when map needs to release unused image resources
+   */
+  public void removeOnRemoveUnusedStyleImagesListener(@NonNull OnRemoveUnusedStyleImagesListener listener) {
+    boolean lastListenerRemoved = mapChangeReceiver.removeOnRemoveUnusedStyleImagesListener(listener);
+    if (lastListenerRemoved) {
+      addDefaultRemoveUnusedStyleImagesListener();
+    }
+  }
+
+  private void addDefaultRemoveUnusedStyleImagesListener() {
+    if (defaultOnRemoveUnusedStyleImagesListener == null) {
+      // Initialize default callback for OnRemoveUnusedStyleImages notifications.
+      defaultOnRemoveUnusedStyleImagesListener = new OnRemoveUnusedStyleImagesListener() {
+        @Override
+        public void onRemoveUnusedStyleImages(@NonNull String[] ids) {
+          if (mapboxMap != null && mapboxMap.getStyle() != null) {
+            for (String id : ids) {
+              mapboxMap.getStyle().removeImage(id);
```

Comment:
NIT: could you create a local reference to the `Style` before the if-check to deal with the lint warning?